NetKVM/Common/ParaNdis_TX.cpp

```diff
@@ -896,6 +936,16 @@ void CNB::DoIPHdrCSO(PVOID IpHeader, ULONG EthPayloadLength) const
 
 bool CNB::FillDescriptorSGList(CTXDescriptor &Descriptor, ULONG ParsedHeadersLength) const
 {
+#if NDIS_SUPPORT_NDIS650
+    #if ALICLOUD_MAX_FRAGMENTS_IN_ONE_NB
+        DPrintf(0, "zjm: [%s] ParsedHeadersLength [%d] NET_BUFFER_DATA_OFFSET(m_NB) [%d] ++  [%d] m_bCopyMode [%d]\n", __FUNCTION__, ParsedHeadersLength, NET_BUFFER_DATA_OFFSET(m_NB), ParsedHeadersLength + NET_BUFFER_DATA_OFFSET(m_NB), m_bCopyMode);
+        if (m_bCopyMode)
+        {
+            return Descriptor.SetupHeaders(ParsedHeadersLength) &&
+                MapCopyDataToVirtioSGL(Descriptor, ParsedHeadersLength);
```

Comment:
Why NET_BUFFER_DATA_OFFSET(m_NB) is not added? Is there by intention?

Reply:
@ybendito ,yes, by intention. Because m_CopySGL is different from m_SGL; its data segment does not have such an offset.